cmd/common/utils/trace.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+// BaseTraceConfig is used to contain information used to manage a trace.
+type BaseTraceConfig struct {
```

Comment:
I think this is too related to the CRD. I know local-gadget is reuses many of the logic around the trace, but I think this is something that we'll reduce in the future and this PR is a good opportunity not to include more logic related to it here.

My specific point is: I think this `BaseTraceConfig` shouldn't be the base for the trace config used in kubectl-gadget. It's possible that they share some info, but I'd like to have them as separated as possible.

Reply:
You are right. Given that we are planning to make `local-gadget` independent from the CRD, it is better to keep that code separated so that when we introduce such a change, there will be no need to modify `kubectl-gadget` or `common` packages but only `local-gadget` itself.